edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -201,6 +200,16 @@ public Object doInHibernate(Session session) throws HibernateException {
 			log.warn("AUTHORIZATION FAILURE: User " + getUserUid() + " in gradebook " + gradebookUid + " attempted to get assignment " + assignmentName);
 			throw new SecurityException("You do not have permission to perform this operation");
 		}
+		if (NumberUtils.isNumber(assignmentName)) {
+			try {
+				return getAssignment(gradebookUid, NumberUtils.toLong(assignmentName, -1L));
+			}
+			//Continue on if the assignment is not found
+			catch (AssessmentNotFoundException e) {
+				log.debug("{}",e.getMessage());
```

Comment:
Ordinarily this should have a message, but AssessmentNotFoundException only takes a message anyway.